### PR TITLE
refactor rebuild creation and create function for freetype

### PR DESCRIPTION
**Pull request by CJ-Wright**
Pending freetype pin see: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/215

### CODE REVIEW

**Comment by ocefpaf** [comment]
> Pending freetype pin see: [conda-forge/conda-forge-pinning-feedstock#215](https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/215)

The freetype pinning is not happening. Can we drop the migrator line and merge this one? I'm happy to do the changes if you agree.

**Comment by ocefpaf** [comment]
Forgot to xref the freetype discussion: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/215#issuecomment-481750983

File: conda_forge_tick/auto_tick.xsh [hunk 1]
```diff
@@ -447,7 +447,6 @@ def initialize_migrators(do_rebuild=False):
     add_rebuild_openssl($MIGRATORS, gx)
     add_rebuild_libprotobuf($MIGRATORS, gx)
     add_rebuild_blas($MIGRATORS, gx)
-    add_rebuild_successors($MIGRATORS, gx, 'freetype', '2.10')
```
Comment (ocefpaf) [inline conda_forge_tick/auto_tick.xsh]: Hope that is OK. Waiting on this one to issue the `proj4` migration PR.
Ping @xhochy who will be doing something similar for https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/218

**Comment by CJ-Wright** [comment]
LGTM

**Pull request merged**